src/main/java/org/spongepowered/api/scoreboard/objective/Objective.java

```diff
@@ -0,0 +1,132 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.scoreboard.objective;
+
+import com.google.common.base.Optional;
+import org.spongepowered.api.scoreboard.Score;
+import org.spongepowered.api.scoreboard.Scoreboard;
+import org.spongepowered.api.scoreboard.displayslot.DisplaySlot;
+import org.spongepowered.api.scoreboard.objective.displaymode.ObjectiveDisplayMode;
+
+import javax.annotation.Nullable;
+
+/**
+ * An objective tracks an integer score for each entry it contains.
+ *
+ * <p>Entries can be updated by plugins, by in-game commands, or automatically by the game, depending
+ * on their {@link org.spongepowered.api.scoreboard.critieria.Criterion}.</p>
+ */
+public interface Objective {
+
+    /**
+     * Gets the name of this Objective.
+     *
+     * @return The objective's name
+     */
+    String getName();
+
+    /**
+     * Gets the name displayed to players.
+     *
+     * @return The objective's display name
+     */
+    String getDisplayName();
+
+    /**
+     * Sets the name displayed to players.
+     *
+     * @param displayName Display name to set
+     * @throws IllegalArgumentException if displayName is longer than 32
+     *     characters.
+     */
+    void setDisplayName(String displayName) throws IllegalArgumentException;
+
+    /**
+     * Gets the criteria for this objective.
+     *
+     * @return This objective's criteria
+     */
+    String getCriteria();
+
+    /**
+     * Gets the scoreboard to which this objective is attached, if present.
+     *
+     * @return Owning scoreboard, if present
+     */
+    Optional<Scoreboard> getScoreboard();
+
+    /**
+     * Sets the scoreboard to which this objective is attached.
+     *
+     * @param scoreboard The scoreboard
+     */
+    void setScoreboard(Scoreboard scoreboard);
+
+    /**
+     * Sets this objective to display on the specified slot for the
+     * scoreboard, removing it from any other display slot.
+     *
+     * If another objective is set to the same display slot, that objective will
```

Comment:
Should be either wrapped in `<p>` and `</p>` or on the same paragraph as the previous one